test/sh_do_test.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+if [ "${RESULTS_DIR}" == "" ]; then
+    echo Note: this program is normally called through the Makefile, it
+    echo is not meant to be called directly by the user.
+    exit 1
+fi
+
+# TEST_DIR should be one of compilable, fail_compilation or runnable
+export TEST_DIR=$1
+export TEST_NAME=$2
+script_name=${TEST_DIR}/${TEST_NAME}.sh
+
+echo " ... ${script_name}"
+
+output_file=${RESULTS_DIR}/${script_name}.out
+rm -f ${output_file}
+
+./${script_name} > ${output_file}  2>&1
```

Comment:
A future extension could use `source` here.
Advantages:
- use a trap, to show the error of the failure
- activate `set -ue -o pipefail` by default
- activate `set +x` by default, redirect the debug log into a `XYZ.debug` file and only show it when there's a failure.